### DISCUSSION TITLE
Sort Function

### DISCUSSION THREAD

**Discussion by mohbusaidy** [neuefische/questions#50] [Web] [answered]
Hey Leute ich hab meine Sort() implementiert mit einem UseEffect, und habe das State Up Lifting auch als Methode genutzt. Ich weiß leider nicht wo ich genau falsch gegangen bin, weil die console mir auch unschlüssig bleibt. 

Hier der Link zu meiner [Branch ](https://github.com/mohbusaidy/Capstone-Project/tree/favorites)

**Comment by actyvystom** [comment]
Hallo Mo, schau doch bitte mal hier: [How-to-ask-good-questions](https://github.com/neuefische/questions/wiki/How-to-ask-good-questions) und versuche deine Frage exakter zu formulieren. Das macht es allen einfacher. 
Danke Dir!

**Comment by CJAkkuec** [comment, marked as answer]
Was @actyvystom sagt, und zudem noch:
```
useEffect(() => {
    if (sortProducts === 'Low to high') {
      const lth = products.price((a, b) => a - b);
      setSortProducts(lth);
    }
    if (sortProducts === 'High to low') {
      const htl = products.price((a, b) => a - b).reverse();
      setSortProducts(htl);
    }
  });
```
... auf den ersten Blick könntest du hier auch `const htl = products.price((a, b) => b - a)` schreiben.

Edit: Oh, und du solltest noch `products` und `sortProducts` als dependency für deinen useEffect hinzufügen, sonst könnte es evtl. ein bisschen arg rattern.

**Reply by mohbusaidy** [reply]
kann ich dann quasi einfach zwei const schreiben?

**Reply by CJAkkuec** [reply]
> kann ich dann quasi einfach zwei const schreiben?

Bin mir nicht sicher, was du damit meinst. : )


Das hier wäre z.B. auch eine Möglichkeit

  ```
const [sortedLowToHigh, setSortedLowToHigh] = useState(true);

const sortedArray = initialArray.sort((a, b) => {
    if (sortedLowToHigh) {
      return a - b ;
    }
    else {
      return b - a ;
    }
  });
```
In `initialArray` steckt dein ursprünglicher Array, der ist deine "universal source of truth". In `sortedArray` steckt dann der neue, sortierte Array, denn du an die Stelle packen kannst, wo du ihn brauchst.

Wenn du nur zwei Optionen zum Sortieren hast, kannst du das über einen Boolean Value machen.
Evtl. muss du schauen, ob du an den Wert nach dem du sortierst nur über dot notation rankommst.
Also sowas wie return `a.value - b.value`.

**Reply by mohbusaidy** [reply]
Vielen dank, durch den ansatz dieser funktion hab ich das lösen können!
